### PR TITLE
Add support to create PowerShell 7.0 and Java 11 function apps

### PR DESCRIPTION
**Pull request by Francisco-Gamino**
## Description

Add support to create PowerShell 7.0 and Java 11 function apps. This PR addresses:
https://github.com/Azure/azure-powershell/issues/11956
https://github.com/Azure/azure-powershell/issues/11957

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by Francisco-Gamino** [comment]
cc: @anirudhgarg @anthonychu @eamonoreilly @amamounelsayed

File: src/Functions/Az.Functions.psd1 [hunk 1]
```diff
@@ -1,7 +1,7 @@
 @{
   GUID = 'eafced71-8742-4a2c-5afd-13117428dd90'
   RootModule = './Az.Functions.psm1'
-  ModuleVersion = '1.0.0'
+  ModuleVersion = '1.0.1'
```
Comment (dingmeng-xue) [inline src/Functions/Az.Functions.psd1:4]: Module version doesn't need to be changed.

Comment (Francisco-Gamino) [reply src/Functions/Az.Functions.psd1:4]: @dingmeng-xue should I revert this file? Or keep it as is? 

Please note that I have also updated `src/Functions/readme.md` to increase the module version to `1.0.1`.

File: src/Functions/custom/HelperFunctions.ps1 [hunk 2]
```diff
@@ -51,13 +51,13 @@ $constants["RuntimeToDefaultOSType"] = @{
     'Python' = 'Linux'
 }
 
-# This are use for tab completion for the RuntimeVersion parameter in New-AzFunctionApp
+# These are use for tab completion for the RuntimeVersion parameter in New-AzFunctionApp.
```
Comment (dolauli) [inline src/Functions/custom/HelperFunctions.ps1:54]: A typo. should be "**used**"

File: src/Functions/Az.Functions.psd1 (hunk 1 above)
Comment (Francisco-Gamino) [reply src/Functions/Az.Functions.psd1:4]: @dingmeng-xue reverted back `src/Functions/Az.Functions.psd1`

File: src/Functions/custom/HelperFunctions.ps1 (hunk 2 above)
Comment (Francisco-Gamino) [reply src/Functions/custom/HelperFunctions.ps1:54]: @dolauli -- Good catch. Fixed.

**Pull request merged**